bundles/binding/org.openhab.binding.mystromecopower/src/main/java/org/openhab/binding/mystromecopower/internal/api/MystromClient.java

```diff
@@ -0,0 +1,304 @@
+					"?authToken=" + this.authToken
+					+ "&id=" + deviceId)
+				.openConnection();
+			
+			httpURLConnection.connect();
+			
+			int responseCode = httpURLConnection.getResponseCode();
+			
+			if (responseCode != HttpURLConnection.HTTP_OK) {
+				logger.error("Get device info http code: '{}'", responseCode);
+				return null;
+			}
+		
+			InputStream inputStream = httpURLConnection.getInputStream();
+			reader = new InputStreamReader(inputStream, "UTF-8");
+			JsonObject jsonObject = (JsonObject) jsonParser.parse(reader);
+			
+			String status = jsonObject.get("status").getAsString();
+			if(!status.equals("ok")){
+				logger.error("Error while getting device info id: '{}' status '{}'", deviceId, status);
+				return null;
+			}
+			
+			GetDeviceInfoResult result = gson.fromJson(jsonObject, GetDeviceInfoResult.class);
+			
+			return result.device;
+		}
+		catch(Exception ex){
+			logger.error("Error getting device info  with id: '{}', detail '{}'", deviceId, ex.toString());
+			return null;
+		} finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException ignored) {
+
+				}
+			}
+		}
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#ChangeState(java.lang.String, java.lang.Boolean)
+	 */
+	@Override
+	public Boolean ChangeState(String deviceId, Boolean newStateIsOn) {
+		Reader reader = null;
+		logger.debug(
+			"Change state for device id '{}', new state is on: '{}'",
+			deviceId,
+			newStateIsOn);
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "device/switch" +
+					"?authToken=" + this.authToken
+					+ "&id=" + deviceId
+					+ "&on=" + newStateIsOn.toString())
+				.openConnection();
+		
+			InputStream inputStream = httpURLConnection.getInputStream();
+			reader = new InputStreamReader(inputStream, "UTF-8");
+			JsonObject jsonObject = (JsonObject) jsonParser.parse(reader);
+			
+			String status = jsonObject.get("status").getAsString();
+			if(!status.equals("ok")){
+				String error = jsonObject.get("error").getAsString();
+				logger.error("Unable to switch state for device '{}' error '{}'", deviceId, error);
+				return false;
+			}
+			
+			String newState = jsonObject.get("state").getAsString();
+			logger.debug(
+				"Switch state for device '{}' successfull, state is '{}'",
+				deviceId,
+				newState);
+			
+			return true;
+		}
+		catch(Exception ex)
+		{
+			logger.error("Error set state: '{}'", ex.toString());
+			return false;
+		} 
+		finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException ignored) {
+
+				}
+			}
+		}
+	}
+	
+	@Override
+	public void RestartMaster(String deviceId) {
+		Reader reader = null;
+		logger.debug(
+			"Restart master device id '{}'",
+			deviceId);
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "device/restart" +
```

Comment:
Could you please assign the URL to a local variable before opening the connection? This helps while debugging problems.